src/ngx_http_lua_balancer.c

```diff
@@ -299,7 +300,30 @@ ngx_http_lua_balancer_get_peer(ngx_peer_connection_t *pc, void *data)
     if (bp->sockaddr && bp->socklen) {
         pc->sockaddr = bp->sockaddr;
         pc->socklen = bp->socklen;
-        pc->name = &bp->host;
+
+        /* balancer module has no peer actually, it just create one virtual peer
+         * when necessary, 
+         * at the same time, because the peer_name is returned as a pointer, its
+         * content changed after using balancer_by_lua, if we just returned the pointer,
+         * nginx may always use the last peer_name, so we return a copy pointer of
+         * bp->host to pc, in case upstream_addr
+         * always use the last peer name
+         */
+        peer_name = ngx_palloc(r->pool, sizeof(ngx_str_t));
+        if (peer_name == NULL) {
+            return NGX_ERROR;
+        }
+
+        ngx_memzero(peer_name, sizeof(ngx_str_t));
```

Comment:
We do not need this since we explicitly initialize both of the fields of `ngx_str_t` later anyway.